dragonfly/block/note_block.go

```diff
@@ -0,0 +1,87 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/instrument"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/df-mc/dragonfly/dragonfly/world/particle"
+	"github.com/df-mc/dragonfly/dragonfly/world/sound"
+)
+
+// NoteBlock is a musical block that emits sounds when powered with redstone.
+type NoteBlock struct {
+	nbt
+	solid
+	bass
+
+	// Pitch is the current pitch the note block is set to. Value ranges from 0-24.
+	Pitch int32
```

Comment:
Would be best to keep this an `int` for ease of use. Make sure to cast it to an int32 in the NBT when you change it.